packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts

```diff
@@ -88,6 +88,142 @@ type SharedObjCallback = (
 	dataStore: ITestFluidObject,
 ) => void | Promise<void>;
 
+/**
+ * load container, pause, create (local) ops from callback, then optionally send ops before closing container
+ */
+const getPendingOps = async (
+	testContainerConfig: ITestContainerConfig,
+	testObjectProvider: ITestObjectProvider,
+	send: boolean,
+	cb: SharedObjCallback = () => undefined,
+) => {
+	const container: IContainerExperimental =
+		await testObjectProvider.loadTestContainer(testContainerConfig);
+	await waitForContainerConnection(container);
+	const dataStore = (await container.getEntryPoint()) as ITestFluidObject;
+
+	[...Array(lots).keys()].map((i) =>
+		dataStore.root.set(`make sure csn is > 1 so it doesn't hide bugs ${i}`, i),
+	);
+
+	await testObjectProvider.ensureSynchronized();
+	await testObjectProvider.opProcessingController.pauseProcessing(container);
+	assert(dataStore.runtime.deltaManager.outbound.paused);
+
+	await cb(container, dataStore);
+
+	let pendingState: string | undefined;
+	if (send) {
+		pendingState = await container.getPendingLocalState?.();
+		await testObjectProvider.ensureSynchronized(); // Note: This will resume processing to get synchronized
+		container.close();
+	} else {
+		pendingState = await container.closeAndGetPendingLocalState?.();
+	}
+
+	testObjectProvider.opProcessingController.resumeProcessing();
+
+	assert.ok(pendingState);
+	return pendingState;
+};
+
+/**
+ * Load a Container using testContainerConfig and the given testObjectProvider
+ *
+ * @param testObjectProvider - For accessing Loader/Driver
+ * @param request - Request to use when loading
+ * @param pendingLocalState - (Optional) custom PendingLocalState to load from. Defaults to using getPendingOps helper if omitted.
+ * @returns A container instance with a connect function to unblock the Driver (simulating coming back from offline)
+ */
+async function loadOffline(
+	testContainerConfig: ITestContainerConfig,
```

Comment:
since this is const why not just move it out too, so it can be used in the function, rather than needing to pass it

Reply:
In other e2e test files I've needed to adjust it for certain tests, that's why I decided to pull it out.